### ISSUE TITLE
Add information about available msbuild command line options

### ISSUE THREAD

**Issue by aannenko** [dotnet/docs#7568]
"dotnet publish" supports msbuild command line options like /p:Version=1.2.3.4
Please add information about it to this article as well as several examples.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 211afdc7-c796-202d-239d-823f2a499dff
* Version Independent ID: 9439897e-dcb3-bc2e-5cae-35c1ca2e0681
* Content: [dotnet publish command - .NET Core CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-publish?tabs=netcore21)
* Content Source: [docs/core/tools/dotnet-publish.md](https://github.com/dotnet/docs/blob/master/docs/core/tools/dotnet-publish.md)
* Product: **dotnet-core**
* GitHub Login: @mairaw
* Microsoft Alias: **mairaw**

**Comment by Thraka** [comment]
I agree that we should have some information and examples about msbuild, but we shouldn't go too deep. We should however link to more in depth articles about it.

**Comment by satrapu** [comment]
I'm publishing a .NET Core 2.1 app via GitLab and this is how I state the desired version details:
````bash
dotnet publish MyConsoleAppProject.csproj --configuration Release --output .my-console-app --runtime win-x86 --self-contained true /p:VersionPrefix=2.34.12 /p:VersionSuffix=rc1 /p:InformationalVersion=my-console-app-2.34.12-rc1+sha.6afh4vew.win-x86
````

The versioning related build properties have been set to specific values following semantic versioning guidelines (https://semver.org/):
*  /p:VersionPrefix - see https://semver.org/#spec-item-2
* /p:VersionSuffix - see https://semver.org/#spec-item-9
* /p:InformationalVersion - represents the concatenation of VersionPrefix and VersionSuffix plus the build metadata, in my case the first 8 characters of the git commit hash followed by the runtime id (e.g. win-x86); see more about build metadata here: https://semver.org/#spec-item-10

**Comment by Lindsey1986** [comment]
@satrapu If I may ask, how are you generating the full string `my-console-app-2.34.12-rc1+sha.6afh4vew.win-x86`?

**Comment by satrapu** [comment]
> 
> 
> @satrapu If I may ask, how are you generating the full string `my-console-app-2.34.12-rc1+sha.6afh4vew.win-x86`?

I was using a [script](https://docs.gitlab.com/ee/ci/yaml/#script)  step in a GitLab pipeline (a YAML file) in conjunction with some Windows Batch and several pipeline variables to form the `my-console-app-2.34.12-rc1+sha.6afh4vew.win-x86` string. The end-goal was to publish a .NET Core 2.x web application targeting Windows running on 32b bits using the [self-contained deployement](https://docs.microsoft.com/en-us/dotnet/core/deploying/#self-contained-deployments-scd) mechanism.

| Token                          | Description           | Location                                                 |
| ----------------------  | ------------------ | ----------------------------------------  |
| my-console-app          | Application name  | Declared as a variable inside YAML file    |                                      
| 2.34.12                       | Semantic version   | Declared as a variable, outside YAML file |
| rc1                              | Pre-release info     | Declared as a variable, outside YAML file |
| 6afh4vew |  The first 8 characters of the currently built Git commit hash     | Computed by the pipeline at build time  |
| win-x86 | [RID](https://docs.microsoft.com/en-us/dotnet/core/rid-catalog#windows-rids) | Declared as a variable inside YAML file |

The most interesting/hard part was extracting the first 8 characters out of the current Git commit hash (e.g.  `6afh4vew`) - see [here](https://stackoverflow.com/a/636391) how to employ Windows Batch for this task.
Unfortunately, I no longer have access to the original source code to provide more details and currently I'm not using GitLab to discover the exact solution once again.

An alternative to Windows Batch is to invoke a PowerShell script with a bunch of parameters to create the final string.

**Comment by Lindsey1986** [comment]
@satrapu Thank you so much for all the detail! Very useful!

**Comment by sfoslund** [comment]
@tdykstra is there anything left to do here? It looks like this is solved by #17671. The synopsis also contains some examples of msbuild property usage with specific properties that are relevant to publishing: (` [-p:PublishReadyToRun=true] [-p:PublishSingleFile=true] [-p:PublishTrimmed=true]`)

**Comment by tdykstra** [comment]
Yes, I'll close this issue, as the examples added in #17671 show msbuild property usage.
